From 8457d0fd20c476750736267d295e872a130983b6 Mon Sep 17 00:00:00 2001
From: Osmond Chen <osmond.chen@mic.com.tw>
Date: Wed, 16 Mar 2022 09:38:37 +0800
Subject: [PATCH 2/5] take zero to replace first reading to fill buffer

---
 healthMonitor.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/healthMonitor.cpp b/healthMonitor.cpp
index b2424a3..a712b08 100755
--- a/healthMonitor.cpp
+++ b/healthMonitor.cpp
@@ -289,7 +289,7 @@ void HealthSensor::initHealthSensor(const std::vector<std::string>& chassisIds)
     /* Initialize value queue with initial sensor reading */
     for (int i = 0; i < sensorConfig.windowSize; i++)
     {
-        valQueue.push_back(value);
+        valQueue.push_back(0);
     }
 
     /* Initialize unit value (Percent) for utilization sensor */
-- 
2.7.4

