From e3d3878bb71cd99e73daab99128e6b73a4fed73a Mon Sep 17 00:00:00 2001
From: John Chung <john.chung@mic.com.tw>
Date: Tue, 22 Mar 2022 15:40:21 +0800
Subject: [PATCH 18/24] Add bmc self testing command supported

---
 apphandler.cpp | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/apphandler.cpp b/apphandler.cpp
index 1f0e6b4..8e6284d 100755
--- a/apphandler.cpp
+++ b/apphandler.cpp
@@ -705,9 +705,20 @@ auto ipmiAppGetSelfTestResults() -> ipmi::RspType<uint8_t, uint8_t>
     //      [2] 1b = Internal Use Area of BMC FRU corrupted.
     //      [1] 1b = controller update 'boot block' firmware corrupted.
     //      [0] 1b = controller operational firmware corrupted.
-    constexpr uint8_t notImplemented = 0x56;
+    constexpr uint8_t passed = 0x55;
+    // constexpr uint8_t notImplemented = 0x56;
+    constexpr uint8_t deviceCorrupted = 0x57;
     constexpr uint8_t zero = 0;
-    return ipmi::responseSuccess(notImplemented, zero);
+
+    auto busp = getSdBus();
+
+    std::ifstream bootFinish("/run/booting-finish");
+    if(!bootFinish.good())
+    {
+        return ipmi::responseSuccess(deviceCorrupted, zero);
+    }
+
+    return ipmi::responseSuccess(passed, zero);
 }
 
 static constexpr size_t uuidBinaryLength = 16;
-- 
2.7.4

