From b130098d339fe6a3a0d3c7cf9f6dcc8f2dd36142 Mon Sep 17 00:00:00 2001
From: John Chung <john.chung@mic.com.tw>
Date: Tue, 22 Mar 2022 15:50:11 +0800
Subject: [PATCH 21/24] Clear store sensor data when SDR command failed

---
 dbus-sdr/sensorcommands.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dbus-sdr/sensorcommands.cpp b/dbus-sdr/sensorcommands.cpp
index f93c789..1e11d0a 100755
--- a/dbus-sdr/sensorcommands.cpp
+++ b/dbus-sdr/sensorcommands.cpp
@@ -2456,6 +2456,7 @@ ipmi::RspType<uint16_t,            // next record ID
         reinterpret_cast<get_sdr::SensorDataRecordHeader*>(record.data());
     if (!hdr)
     {
+        getSensorTree().clear();
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "ipmiStorageGetSDR: record header is null");
         return ipmi::responseSuccess(nextRecordId, record);
-- 
2.7.4

